src/rebuild/srv.c

```diff
@@ -1886,14 +1979,20 @@ rebuild_task_ult(void *arg)
 		D_DEBUG(DB_REBUILD, DF_RB " rebuild is not done: " DF_RC "\n", DP_RB_RGT(rgt),
 			DP_RC(rgt->rgt_status.rs_errno));
 
-		if (rgt->rgt_abort && rgt->rgt_status.rs_errno == 0) {
-			/* If the leader is stopped due to the leader change,
-			 * then let's do not stop the real rebuild(scan/pull
-			 * ults), because the new leader will resend the
-			 * scan requests, which will then become the new
-			 * leader to track the rebuild.
+		if (rgt->rgt_abort && rgt->rgt_stop_admin) {
+			/* Administrator asked to stop rebuild (more than a leader abort - stop on
+			 * all engines). Notify all engines to stop with
+			 * rebuild_leader_status_notify().
 			 */
-			D_DEBUG(DB_REBUILD, DF_RB " Only stop the leader\n", DP_RB_RGT(rgt));
+			D_INFO(DF_RB ": stop rebuild due to admin command.\n", DP_RB_RGT(rgt));
+		} else if (rgt->rgt_abort && rgt->rgt_status.rs_errno == 0) {
+			/* If the leader rebuild is aborted due to a leader change, then do not
+			 * abort the real rebuild(scan/pull ults), because the new leader will
+			 * resend the scan requests, which will then become the new leader to track
+			 * the rebuild.
+			 */
+			D_DEBUG(DB_REBUILD, DF_RB " abort rebuild on leader only\n",
```

Comment:
the log is incorrect, it indeed did not abort the rebuild, it just stop the leader. the rebuild is keep going ahead.
so the original log looks better.

Reply:
will change the log text and will make sure rgt->rgt_status.rs_errno is set to -DER_OP_CANCELED for that case in this function.